responsibleai/responsibleai/managers/data_balance_manager.py

```diff
@@ -0,0 +1,318 @@
+
+
+class DataBalanceManager(BaseManager):
+    """
+    Defines the DataBalanceManager for finding data balance measures
+    on a dataset.
+    """
+
+    def __init__(
+        self, train: pd.DataFrame, test: pd.DataFrame, target_column: str
+    ):
+        """
+        Creates a DataBalanceManager object.
+
+        :param train: The training data.
+        :type train: pd.DataFrame
+        :param test: The test data.
+        :type test: pd.DataFrame
+        :param target_column: The name of the target column.
+        :type target_column: str
+        """
+        self._train = train
+        self._test = test
+        self._target_column = target_column
+        self._is_added = False
+
+        self._df = None
+        if self._train is not None or self._test is not None:
+            self._df = pd.concat([self._train, self._test])
+
+        # Populated in add()
+        self._cols_of_interest = None
+        self._pos_label = None
+
+        # Populated in compute()
+        self._data_balance_measures = None
+
+    def add(
+        self, cols_of_interest: List[str], pos_label: Optional[str] = None
+    ):
+        """
+        Add data balance measures to be computed later.
+
+        :param cols_of_interest: The names of the columns to be used
+            for computing data balance measures.
+        :type cols_of_interest: List[str]
+        :param pos_label: If the target column does not consist of 0 and 1,
+            the label value that denotes a positive label.
+        :type pos_label: Optional[str]
+        """
+        self._cols_of_interest = cols_of_interest
+        self._pos_label = pos_label
+
+        # Let user see exceptions early in add() before calling compute()
+        self._validate()
+        self._is_added = True
+
+    def _validate(self):
+        """
+        Validate that data balance measures can be computed. Raises ValueError
+        if not valid.
+        """
+        if self._train is None and self._test is None:
+            raise ValueError(
+                (
+                    "Either `train` or `test` must be provided to RAIInsights"
+                    " to compute data balance measures."
+                )
+            )
+
+        if not self._cols_of_interest or not self._target_column:
+            raise ValueError(
+                (
+                    "Both `cols_of_interest` and `target_column` must be"
+                    " provided to compute data balance measures."
+                )
+            )
+
+        if not all(col in self._df.columns for col in self._cols_of_interest):
+            raise ValueError(
+                (
+                    "All columns in `cols_of_interest` must be present in"
+                    " the dataset."
+                )
+            )
+
+        if self._target_column not in self._df.columns:
+            raise ValueError(
+                (
+                    f"The target_column `{self._target_column}` must be"
```

Comment:
It seems like sometimes there's a backquote and sometimes a normal quote (see next error message). Can we do this uniformly? I think normal quotes are fine (?)

Reply:
Ya my mistake! Switched to normal quotes